Lib/bdb.py

```diff
@@ -209,9 +234,15 @@ def stop_here(self, frame):
         if frame is self.stopframe:
             if self.stoplineno == -1:
                 return False
-            return frame.f_lineno >= self.stoplineno
+            if self.trace_opcodes:
+                return self.lasti != frame.f_lasti
```

Comment:
```suggestion
                return frame.f_lasti >= self.lasti
```
For consistency with `self.stoplineno` checks. If I understand correctly, `frame.f_lasti < self.lasti` is not supposed here (and below)

Reply:
`f_lasti` could easily be going smaller (imagine a for loop). The `stoplineno` check here is for `until` command. For `next`, `self.stoplineno` would be `-1` so it's always true. The reason we needed the `f_lasti` checking here is to avoid double stop at the same instruction for both `line` and `opcode` event. We need the `line` event though to trigger breakpoints.

Reply:
Oh, right, thanks for the explanation!